php/post-types/class-block-post.php

```diff
@@ -1040,7 +1040,7 @@ public function save_block( $data ) {
 		// Block category.
 		if ( isset( $_POST['block-properties-category'] ) ) {
 			$category_slug = sanitize_key( $_POST['block-properties-category'] );
-			$categories    = get_block_categories( the_post() );
+			$categories    = get_block_categories( get_post() );
```

Comment:
It looks like [the_post()](https://developer.wordpress.org/reference/functions/the_post/) doesn't return a value, so [get_post()](https://developer.wordpress.org/reference/functions/get_post/) should work.